### ISSUE TITLE
Go To Block Should Break Before Axis Label

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#999]
The "go to" block should be breaking so that the "y:" is in front of its associated input slot—not on the previous line.

![download](https://cloud.githubusercontent.com/assets/6549787/11011511/a32c3696-84b6-11e5-8d95-e2415a2ddede.png)

**Comment by maryfries** [comment]
See also: https://github.com/ericrosenbaum/BeetleBlocks/issues/185

**Comment by jmoenig** [comment]
breaks within blocks are strictly automatic. Some languages don't prefix parameters in a keyword-like fashion, but instead suffix them. There is some internal support for user-defined breaks, which isn't yet exposed in the GUI. I'm thinking about it...

**Comment by cycomachead** [comment]
> There is some internal support for user-defined breaks, which isn't yet exposed in the GUI.

We could try this out by editing the specs of primitives, right?

**Comment by jmoenig** [comment]
Yes! Switch to dev mode, right click on the GOTO block, edit the spec and insert ```%br``` before the "y". There you go!

**Comment by cycomachead** [comment]
:+1:

**Comment by brianharvey** [comment]
So I think this %br thing should have the semantics of \goodbreak rather than of \break.  Jens is unexcited so this is a student project. :-)

**Comment by cycomachead** [comment]
> So I think this %br thing should have the semantics of \goodbreak rather than of \break. Jens is unexcited so this is a student project. :-)

I'm guessing this is a TeX thing, what's the difference?

I'm tagging fix-it-yourself, then renaming that label up for grabs.

**Comment by cycomachead** [comment]
@jmoenig What should the full spec look like? I'm trying this in the dev mode now, and get a `SyntaxError` when I try make the spec `go to x: %n %br y: %n`. Is that wrong?

**Comment by jmoenig** [comment]
that spec looks right to me, and it does work for me. Maybe there are some other changes in your branch?

**Comment by cycomachead** [comment]
Nope, was working on snap.b.e

It seems to happen whenever I select edit... Save in any dev mode inspector for blockSpecs, even if I don't change anything.

**Comment by jmoenig** [comment]
Oh well, you go to edit mode, then you get the ```go to``` block's context menu (not the user features, the actual morphic menu), and set the *block* spec from there. This is for primitives!

**Comment by jmoenig** [comment]
See, there are two kinds of specs: The *actual* block spec that determines the look and function of a block, and the spec the user creates for a custom block, out of which the *actual* one is derived...

**Comment by cycomachead** [comment]
Ah! That spec function, thanks! :smile: 

Ooooh, editing this does not preserve any inputs...good to know!

This is cooooooool, and SUPER useful for this project I'm working. (Which is planning out writing an autogravder in Snap<em>!</em>)
![download](https://cloud.githubusercontent.com/assets/1505907/11652680/e6d9d4f2-9d4d-11e5-95cb-23ced5a61a8f.png)

**Comment by jmoenig** [comment]
Oh, but these edited specs are not going to be preserved in the saved project! If you need such breaks you'll need to insert them into the block dictionary in ```objects.js``` in your fork (or - perhaps better - overload the desired block specs with your own JS-function block in your project)

**Comment by cycomachead** [comment]
Eventually that'd be a good idea. I just need screenshots now, so I'm fine with the current behavior!

This is more prototyping a spec / API than any implementation so they don't need to look perfect yet. :)
